src/Raven.Server/RavenServer.cs

```diff
@@ -995,8 +995,28 @@ private void StartSnmp()
 
         public TcpListenerStatus StartTcpListener()
         {
+            void EnsureTcpPriorityServerUrlsAreListenTo(int defaultPort, HashSet<(string Host, int Port)> valueTuples, TcpListenerStatus tcpListenerStatus)
+            {
+                if (Configuration.Core.TcpPriorityServerUrls != null)
```

Comment:
These guys are meant for external use, not internal one.
Internally, we'll usually bind to `0.0.0.0`, then we'll use the hostname to decide how to connect.

Another issue is that we probably should use `TcpServerUrl.Public`

Reply:
I don't understand what you mean by "...we probably should use TcpServerUrl.Public"
Here I chose to make sure that the hostname that is in the priority list listens to I could throw if they are not part of the {TcpServerUrls U PublicTcpServerUrl}